File: lesson1/collatz.py
def collatz(n):
    res = []
    while n!=1:
        res.append(n)
        if n%2==0:
            n=n//2
        else:
            n=3*n+1
    res.append(1)
    return res

def format(l):
    return ' '.join(str(a) for a in l)

print('n=', end='')
n=int(input())
print(format(collatz(n)))

File: lesson1/collatz.cc
#include <iostream>

void collatz(int n) {
    while (n!=1) {
        std::cout<<n<<" ";
        if (n%2==0) {
            n=n/2;
        } else {
            n=3*n+1;
        }
    }
    std::cout<<1<<std::endl;
}

int main() {
    int n;
    std::cout<<"n=";
    std::cin>>n;
    collatz(n);
}